src/Web/wwwroot/app/global/mainController.js

```diff
@@ -36,6 +32,20 @@ angular
             }
         });
 
+        $scope.waitClick = function (state, callback) {
+            if ($state.includes(state)) {
+              callback();
+            } else {
+                $state.go(state).then(function() {
+                  callback();
+                });
+            }
+        };
+
+        $scope.toggleSideNav = function() {
+            $scope.$broadcast('settingsToggleSideNav');
```

Comment:
Not sure, was trying to blend it in with some steps you had already taken. I'll switch it to that way.